lib/app/services/api_network.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/widgets.dart';
+import 'package:get_it/get_it.dart';
+import 'package:meta/meta.dart';
+import 'package:http/http.dart' as http;
+import 'package:schulcloud/app/app.dart';
+
+import 'storage.dart';
+
+/// A service that offers networking POST and GET requests to the backend
+/// servers. If the user's token is stored in the authentication storage, the
+/// the requests' headers are automatically enriched with the access token.
+@immutable
+class ApiNetworkService {
+  const ApiNetworkService({@required this.apiUrl}) : assert(apiUrl != null);
+
+  final String apiUrl;
+  String _url(String path) {
+    assert(path != null);
+    return '$apiUrl/$path';
+  }
+
+  NetworkService get _network => services.get<NetworkService>();
+
+  Map<String, String> _getHeaders() {
+    final storage = services.get<StorageService>();
+    return {
+      'Content-Type': 'application/json',
+      if (storage.hasToken)
+        'Authorization': 'Bearer ${storage.token.getValue()}',
+    };
```

Comment:
`StorageService` is available as `services.storage`
```suggestion
    return {
      'Content-Type': 'application/json',
      if (storage.hasToken)
        'Authorization': 'Bearer ${services.storage.token.getValue()}',
    };
```